src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -246,6 +251,33 @@ public RevisionsCollectionConfiguration GetRevisionsConfiguration(string collect
                         }
                     }
 
+                    if (fromReplication)
+                    {
+                        void PutFromRevision() => _documentsStorage.Put(context, id, null, document, lastModifiedTicks, changeVector, 
+                            flags & ~DocumentFlags.Revision, nonPersistentFlags | NonPersistentDocumentFlags.FromRevision);
+
+                        try
+                        {
+                            var hasDoc = _documentsStorage.GetTableValueReaderForDocument(context, lowerId, out TableValueReader tvr);
+                            if (hasDoc == false)
+                            {
+                                PutFromRevision();
+                            }
+                            else
+                            {
+                                var docChangeVector = TableValueToChangeVector(ref tvr, (int)DocumentsTable.ChangeVector);
+                                if (changeVector.GreaterThan(docChangeVector))
+                                {
+                                    PutFromRevision();
+                                }
+                            }
+                        }
+                        catch (DocumentConflictException)
```

Comment:
What would throw the conflict, just the `GetTableValueReaderForDocument`, right? In that case, let us move the `try`/`catch` to just wrap it, and exit early. It isn't clear if we are looking at the exception thrown from the `Put` here